### PR TITLE
Add bootstrap and webpacker

### PR DESCRIPTION
**Pull request by anastasiiahryniv**
- [x] fixed webpacker
- [x] added bootstrap
- [x] added User Model (STI Student, Instructor)
- [x] added Courses Model
- [x] added Enrolments Model
- [x] added FactoryBot
- [x] added tests

### CODE REVIEW

File: db/migrate/20240929171251_create_courses.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreateCourses < ActiveRecord::Migration[6.1]
+  create_table "courses", force: :cascade do |t|
+    t.string "name", null: false
+    t.bigint "instructor_id", null: false
+    t.bigint "student_id", null: false
```
Comment (VasylYaremkiv) [inline db/migrate/20240929171251_create_courses.rb:5]: Please remove relation to `student`

File: app/models/course.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Course < ApplicationRecord
+  belongs_to :instructor, class_name: 'Instructor'
```
Comment (VasylYaremkiv) [inline app/models/course.rb:2]: Please add status

File: app/models/enrollment.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Enrollment < ApplicationRecord
+  belongs_to :student, class_name: 'Student'
```
Comment (VasylYaremkiv) [inline app/models/enrollment.rb:2]: Please add `status` and `evaluation`

File: app/models/course.rb (hunk 2 above)
Comment (VasylYaremkiv) [reply app/models/course.rb:2]: Add required `name` field

File: app/models/course.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+  has_many :enrollments, dependent: :destroy
+  has_many :students, through: :enrollments
+
```
Comment (VasylYaremkiv) [inline app/models/course.rb:5]: Please group methods. e.g.:

```
class Course
  # Constants
  MAX_NAME_LENGH = 500

  # Associations
  belongs_to :instructor, class_name: 'Instructor'
  has_many :enrollments, dependent: :destroy
  has_many :students, through: :enrollments
  
  # Validations
  validates :name, presence: true, length: { maximum: MAX_NAME_LENGH }
  
  # Enrollments
  enum status: { draft: 10, active: 20, inactive: 30 }
end
```

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class User < ApplicationRecord
+  validates :name, :surname, :email, presence: true
```
Comment (VasylYaremkiv) [inline app/models/user.rb:2]: Please split each field to separate validation

File: test/models/user_test.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (VasylYaremkiv) [inline test/models/user_test.rb:4]:
```
let(:user_params) { { name: 'John', surname: 'Doe', email: 'john.doe@test.com' } }

test 'user should be valid' do
  user = User.new(user_params)
  assert user.valid?
end

test 'name should be present' do
  user = User.new(user_params.except(:name))
  assert_not user.valid?
end
```

File: app/models/instructor.rb [hunk 7]
```diff
@@ -0,0 +1,3 @@
+class Instructor < User
```
Comment (VasylYaremkiv) [inline app/models/instructor.rb:1]: Suggest to separate Instructor and Student without STI

**Review by VasylYaremkiv — CHANGES_REQUESTED** [review]
PTAL

File: db/migrate/20240929171251_create_courses.rb (hunk 1 above)
Comment (anastasiiahryniv) [reply db/migrate/20240929171251_create_courses.rb]: done

**Review by VasylYaremkiv — APPROVED** [review]

**Pull request merged**